### DISCUSSION TITLE
Row select action includes full item

### DISCUSSION THREAD

**Discussion by alexjennings** [table-library/react-table-library#10] [Ideas]
Hi @rwieruch , I have been trying this out for a new project and really enjoying it so far!  I've hit a slight problem where I want to use the row select where the action includes all of the item data for the row instead of just the `id`.  The `id` would be simple to look up for a simple array but I would like to combine this with the tree layout which means traversing through the `node` arrays.

I'm guessing a change would need to be made here https://github.com/table-library/react-table-library/blob/master/src/select/useRowSelect.js#L48 so would you consider a PR or do you feel it would add too much complexity?  Just wanted to get your feeling before I start work on one!

**Comment by rwieruch** [comment]
There is a findNodeById helper function to retrieve a node by id from a tree:

```
import { findNodeById } from '@table-library/react-table-library/common';

const node = findNodeById(nodes, id);
```

You could try this one first.

However, you are right that this is inefficient, because we could pass the node right away. If you are going to propose the change as a PR for the select feature, I'd say we would have to add it to `useTree` as well. Both are using the underlying useIdReducer.js file. Let me know what you think!

Glad that you enjoy working with the library :)
